### DISCUSSION TITLE
Sonoff RF Bridge Command Limit

### DISCUSSION THREAD

**Discussion by ZerGo0** [arendst/Tasmota#16896] [Q&A] [answered]
As stated in the docs a Sonoff RF Bridge is only capable of learning 16 commands. Most outlets have a on and off command which means that I would only be able to use 1 bridge for 8 devices. Is this a hardware limit or a firmware one? I would imagine that it doesn't take a lot of storage to save the commands themselves. I'm not really sure about the computational requirements while receiving and transmitting the RF signals tho.

I've read that you can make your own RF Bridge with a few parts which might be cheaper and would possibly offer more freedom? (Sonoff v2.2 seems to have limited support for different RF signals) Would the same 16 command also apply for selfmade ones?

**Comment by barbudor** [comment, marked as answer]
This is a tasmota limitation as each RfCode takes up to 9 bytes in flash settings and we have to share this flash between all Tasmota features.
Room in flash settings is going low and expanding this is unlikely to occur.

Indeed this limitation is only related to Sonoff RF Bridge
If you setup a DIY device with for example a ESP32/ESP32C3, a RF module and using the RC-Switch driver, it could be possible to expand that further. May be with the help of Berry scripting.
